src/acl/external/unix_group/check_group.cc

```diff
@@ -244,6 +244,7 @@ main(int argc, char *argv[])
             SEND_ERR("");
         }
     }
+    xfree(grents);
```

Comment:
let s keep it ;-)

Reply:
> let s keep it ;-)

Indeed. @rousskov until your refactoring idea actualy takes place this is required to prevent leaks.

Reply:
There should be another at the switch default case exit() on line 187 as well.

Reply:
I would say exit with EXIT_FAILURE and return are two different cases, I would say not worth "fixing" for the former case otherwise better going through a proper goto handling all over. But you can always commit if you disagree I do not mind.

Reply:
We do not need to include "memory still reachable at program exit time" in the definition of a "memory leak". Leaving such memory intact is mostly harmless, and the effort to clean it is often not worth the benefits, especially in C-like code.

IMO, removing this xfree() is the best solution because it keeps PR focused and leaves the code consistent -- `grents` are not explicitly freed while exiting the program; not here, not anywhere else. However, I am still OK with this inconsistency and also do not object to making the code more consistent by adding at least one more xfree() that Amos was talking about. This is a very minor issue not worth fighting about.